src/pages/HomePage/HomePage.tsx

```diff
@@ -1,120 +1,266 @@
-import { useMemo, useReducer, useRef, useState } from "react";
+import ReactQuill from "react-quill";
+import React, { useEffect, useMemo, useReducer, useRef, useState } from "react";
 import { debounce } from "src/utils/Debounce";
-import DOMPurify from 'dompurify';
-import { FloatAddButton, CreateButton } from "src/components/UI/Buttons/Buttons";
-import { BiXCircle } from "src/components/UI/Icons/Icons";
-import { FullPageModal } from "src/components/UI/Modal/FullPageModal/FullPageModal";
+import DOMPurify from "dompurify";
+import {
+	FloatAddButton,
+	CreateButton,
+} from "src/components/UI/Buttons/Buttons";
+import {
+	BiCheck,
+	BiXCircle,
+} from "src/components/UI/Icons/Icons";
 import { useModal } from "src/context/ModalProvider";
+import useAxios from "src/customhook/useAxios";
+import { VAR_ENCODE_TOKEN, VAR_NotPinnedNotes } from "src/utils/Route";
+import StopPropogation from "src/utils/StopPropogation";
+import { preperation } from "src/assets/holders/holders";
+import { useNotes } from "src/context/NotesContext";
+
+const AllNotes = React.lazy(() => import("./AllNotes/AllNotes"));
 
 const ContentDetail = (state, action) => {
-  switch (action.type) {
-    // eslint-disable-next-line no-lone-blocks
-    case "htmlbody": {
-      return { ...state, htmlbody: action.data }
-    };
-    case "title": {
-      return { ...state, title: action.data }
-    };
-    case "color": {
-      return { ...state, color: action.data }
-    };
-    default: {
-      return { ...state };
-    }
-  }
-}
+	console.log(state, action);
+	switch (action.type) {
+		// eslint-disable-next-line no-lone-blocks
+		case "htmlbody": {
+			return { ...state, htmlbody: action.data };
+		}
+		case "title": {
+			return { ...state, title: action.data };
+		}
+		case "color": {
+			return { ...state, color: action.data };
+		}
+		case "priority": {
+			return { ...state, priority: action.data };
+		}
+		case "label": {
+			return { ...state, label: action.data };
+		}
+		case "reset": { 
+			return { ...action.data };
+		}
+		default: {
+			return { ...state };
+		}
+	}
+};
 
+var initialStateNote = {
+	title: "",
+	htmlbody: "",
+	createdOn: "",
+	color: "#bebdff",
+	priority: "",
+	label: "",
+};
 
 export default function HomePage() {
-  const [showNote, setShowNote] = useState(true);
-  const [noteState, noteDispatch] = useReducer(ContentDetail,
-    {
-      title: "",
-      htmlbody: "",
-      createdOn: "",
-      color: "",
-    });
-  var modules = useRef({});
-  const { modalToggle, setmodalToggle } = useModal();
+	const [showNote, setShowNote] = useState(false);
+	const [noteState, noteDispatch] = useReducer(ContentDetail,initialStateNote);
+	const [noteDataSet, SetNoteDataSet] = useNotes();
+	var modules = useRef({});
+	const [response, error, loading, axiosRequest] = useAxios();
+	const { modalToggle, setmodalToggle } = useModal();
+
+	useMemo(() => {
+		modules.current = {
+			toolbar: [
+				[{ header: [1, 2, false] }],
```

Comment:
You should memoize useRef values. Values in useRef persist and doesn't get affected with re renders